zetaclient/btc_signer.go

```diff
@@ -279,18 +279,31 @@ func (signer *BTCSigner) TryProcessOutTx(send *types.CrossChainTx, outTxMan *Out
 		return
 	}
 
-	// FIXME: config chain params
+	// Check receiver P2WPKH address
 	addr, err := btcutil.DecodeAddress(params.Receiver, config.BitconNetParams)
 	if err != nil {
 		logger.Error().Err(err).Msgf("cannot decode address %s ", params.Receiver)
 		return
 	}
+	if !addr.IsForNet(config.BitconNetParams) {
+		logger.Error().Msgf("address %s is not for network %s", params.Receiver, config.BitconNetParams.Name)
+		return
+	}
 	to, ok := addr.(*btcutil.AddressWitnessPubKeyHash)
 	if err != nil || !ok {
 		logger.Error().Err(err).Msgf("cannot convert address %s to P2WPKH address", params.Receiver)
 		return
 	}
 
+	// Add 1 satoshi/byte to gasPrice to avoid minRelayTxFee issue
+	networkInfo, err := signer.rpcClient.GetNetworkInfo()
+	if err != nil {
+		logger.Error().Err(err).Msgf("cannot get bitcoin network info")
+		return
+	}
+	satPerByte := feeRateToSatPerByte(networkInfo.RelayFee)
+	gasprice.Add(gasprice, satPerByte)
```

Comment:
How much significant is the difference?
If it can be significantly higher, it could occur bigger fees to the TSS not paid by the sender and in this case we should consider a more long term solution

Reply:
The number is 1. As of Bitcoin Core 0.9, a minimum fee (currently 1,000 satoshis) has been required to broadcast a transaction across the network.